core/server/services/auth/api-key/admin.js

```diff
@@ -90,10 +91,13 @@ const authenticate = (req, res, next) => {
         // https://github.com/auth0/node-jsonwebtoken/issues/208#issuecomment-231861138
         const secret = Buffer.from(apiKey.get('secret'), 'hex');
 
-        // @TODO When v3 api hits we should check against the api actually being used
+
+        const {pathname} = url.parse(req.originalUrl);
+        const [hasMatch, version = 'v2', api = 'admin', resource = ''] = pathname.match(/ghost\/api\/([^/]+)\/([^/]+)\/(.+)*/);
+
         // ensure the token was meant for this api
         const options = Object.assign({
-            audience: '/v2/admin/'
+            audience: `\/?${version}\/${api}(?:\/${resource})?\/?$`
```

Comment:
Why is the `resource` bit being checked? The check didn't use to be this granular before :thinking:

Reply:
@gargol [Ref](https://github.com/TryGhost/Ghost/pull/10976/files#diff-4b88c110999726b957ca4b3cc23a04feL93):The api key auth was only setup as v2 came into existence, and the token's `audience` verification was hardcoded to use `v2/admin` since we knew this shared middleware will only be triggered for `v2/admin`. 

With now canary/v3 coming in and using same middleware for token auth, it needs to be flexible to be able compare the intended audience with the actual url its trying to hit.

Reply:
Don't mind adding it but let's make it as a follow-up commit. Think PR should be focused on one thing as changes like this might get lost in the context :wink:

Reply:
Sure, I am ok either way, this change is already a separate commit in this PR so at the time of rebase it will reflect as a separate commit on master as well to track it down ;)